src/platform/baytrail/platform.c

```diff
@@ -58,6 +59,17 @@ static const struct sof_ipc_fw_ready ready
 #endif
 		.tag = SOF_TAG,
 		.abi_version = SOF_ABI_VERSION,
+		.compiler = {
+			.micro = CC_MICRO,
+			.minor = CC_MINOR,
+			.major = CC_MAJOR,
+			.name = CC_NAME,
+			.name[0] = CC_NAME[0],
+			.name[1] = CC_NAME[1],
+			.name[2] = CC_NAME[2],
+			.name[3] = CC_NAME[3],
+			.optimization = CC_OPTIMIZATION,
```

Comment:
name is being assigned twice here. It would be better to add a new macro that does the .name[x]=CC_NAME[x]

Reply:
I have written new macro:

```
#define CC_NAME_COPY(field) \
	field = CC_NAME, \
	field[CC_NAME_CAPACITY - 1] = 0
```

It has advantages of both previous methods - is flexible and secure.